src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -467,6 +467,7 @@ void Unit::Update(uint32 p_time)
 
     UpdateSplineMovement(p_time);
     i_motionMaster->UpdateMotion(p_time);
+    UpdateUnderwaterState(GetMap(), GetPositionX(), GetPositionY(), GetPositionZ());
```

Comment:
Seems like it's pretty pointless considering it's already called in Unit::UpdatePosition